dlt/helpers/streamlit_app/dashboard.py

```diff
@@ -0,0 +1,66 @@
+import os
+import sys
+
+import dlt
+import streamlit as st
+from dlt.cli import echo as fmt
+from dlt.helpers.streamlit_app.blocks.query import maybe_run_query
+from dlt.helpers.streamlit_app.blocks.table_hints import list_table_hints
+from dlt.helpers.streamlit_app.menu import menu
+from dlt.helpers.streamlit_app.widgets import schema_picker
+from dlt.pipeline import Pipeline
+
+
+def write_data_explorer_page(
+    pipeline: Pipeline,
+    schema_name: str = None,
+    example_query: str = "",
+    show_charts: bool = True,
+) -> None:
+    """Writes Streamlit app page with a schema and live data preview.
+
+    #### Args:
+        pipeline (Pipeline): Pipeline instance to use.
+        schema_name (str, optional): Name of the schema to display. If None, default schema is used.
+        example_query (str, optional): Example query to be displayed in the SQL Query box.
+        show_charts (bool, optional): Should automatically show charts for the queries from SQL Query box. Defaults to True.
+
+    Raises:
+        MissingDependencyException: Raised when a particular python dependency is not installed
+    """
+
+    st.subheader("Schemas and tables", divider="rainbow")
+    schema_picker(pipeline)
+
+    tables = sorted(
+        st.session_state["schema"].data_tables(),
+        key=lambda table: table["name"],
+    )
+
+    list_table_hints(pipeline, tables)
+    maybe_run_query(
+        pipeline,
+        show_charts=show_charts,
+        example_query=example_query,
+    )
+
+
+def display(pipeline_name: str) -> None:
+    pipeline = dlt.attach(pipeline_name)
+    if pipelines_dir := os.getenv("DLT_PIPELINES_DIR"):
+        pipeline.pipelines_dir = pipelines_dir
+
+    st.session_state["pipeline_name"] = pipeline_name
```

Comment:
why we use session to keep it? previous version had just one `display` method and could write a selected page. and each page had the same interface that was accepting the pipeline instance.